articles/clip-connecting-texts-and-images/index.md

```diff
@@ -0,0 +1,121 @@
+author: willies-ogola
+date: 2021-04-21T00:00:00-16:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/clip-connecting-texts-and-images/hero.jpg
+    alt: CLIP example image
+---
+Contrastive Language-Image Pre-Training (CLIP) is a learning method developed by OpenAI that enables models to learn visual concepts from natural language supervision. This model's main idea is to take images and texts and connect them in a non-generative way.
+<!--more-->
+
+### Table of contents
+
+1. [Prerequisites](#prerequisites)
+2. [Introduction](#introduction)
+3. [Approach](#approach)
+4. [Results](#results)
+5. [Key takeaways](#key-takeaways)
+6. [Wrapping Up](#wrapping-up)
+7. [References](#references)
+
+### Prerequisites
+
+Before reading this article, the reader needs to be familiar with transformers in Natural Language Processing (NLP), Convolutional Neural Networks (CNN), and the working of the novel self-supervised learning (SSL) algorithm. 
+
+### Introduction
+
+Contrastive Language-Image Pre-Training (CLIP) is a learning method developed by OpenAI that enables models to learn visual concepts from natural language supervision. This model's main idea is to take images and texts and connect them in a non-generative way. It isn't entirely a novel idea, but many image-text pair ideas in the past have inspired the CLIP model. There have been papers before predicting the caption of images, but most models don't work too well. 
+A few examples of these models which inspired the CLIP model include the [ConVIRT](https://arxiv.org/pdf/2010.00747.pdf) model, which learns visual representations of medical images by exploiting the naturally occurring pairing of images and textual data, and the [VirTex](https://arxiv.org/pdf/2006.06666.pdf) model, which jointly trained an image CNN and text transformer to predict the captions in visual representations.
+
+So, how does the CLIP model connect images and texts?
+
+### Approach
+
+![The CLIP model](/engineering-education/clip-connecting-texts-and-images/clip-model.PNG)
+
+*[Image Source: OpenAI](https://cdn.openai.com/papers/Learning_Transferable_Visual_Models_From_Natural_Language_Supervision.pdf)*
+
+1. Contrastive pre-training
+
+Contrastive pre-training involves training an image encoder and a text encoder in the multi-modal embedding space to predict the correct pairings of a batch of (image, text) training examples.
+
+#### The Image Encoder
+
+Image augmentation is first performed on the images before being passed into the image encoder module. 
+
+Two different architectures are used for the image encoder: the ResNet-50 and the Vision Transformer (ViT). In the first architecture, The ResNet-50 is used as the base image encoder architecture due to its widespread use. The global average pooling layer present in the ResNet-50 was replaced with an attention pooling mechanism popularly used in NLP to compose representations.
```

Comment:
Some introductory info about the ResNet-50 architecture would better the understanding of a reader who is encountering this for the first time.